src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -55,13 +55,15 @@ public bool IsSuppressed (int id, MessageOrigin warningOrigin, out SuppressMessa
 			if (provider is not IMemberDefinition member)
 				return false;
 
-			MethodDefinition? userDefinedMethod = _context.CompilerGeneratedState.GetUserDefinedMethodForCompilerGeneratedMember (member);
-			if (userDefinedMethod == null)
-				return false;
-
-			Debug.Assert (userDefinedMethod != provider);
+			MethodDefinition? owningMethod;
+			while (_context.CompilerGeneratedState.TryGetOwningMethodForCompilerGeneratedMember (member, out owningMethod)) {
+				Debug.Assert (owningMethod != provider);
+				if (IsSuppressed (id, owningMethod, out info))
+					return true;
+				member = owningMethod;
+			}
```

Comment:
Same thing here as above, are the while loop and recursion redundant here?

Reply:
This one isn't actually recursive, it's calling a different method with the same name.